Catch2 does not compile with clang-8 on macOS

**Describe the bug**
Catch2 does not compile with LLVM clang-8 on macOS. I get the following error message:

```
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/sys/sysctl.h:83,
                 from /Users/torfinn/.conan/data/Catch2/2.7.0/catchorg/stable/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/include/catch2/catch.hpp:8405,
                 from /Users/torfinn/.conan/data/<...>/stable/package/5926da2a8f608c60dbd4ba8eb9c5b650bd6fbdcb/src/main.cpp:18:
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/sys/ucred.h:94:2: error: '_Atomic' does not name a type
```

**Expected behavior**
The program should compile with clang on macOS.

**Reproduction steps**
On macOS, install LLVM with `brew install llvm`.
Compile a main file with clang:

```
#define CATCH_CONFIG_MAIN
#include <catch2/catch.hpp>
```

**Platform information:**
<!-- Fill in any extra information that might be important for your issue. -->
 - OS: **macOS 10.14.3**
 - Compiler+version: **Clang v8.0.0**
 - Catch version: **v2.7.0**
 - XCode **10.2**

**Additional information**
It seems clang doesn't like the macOS system header `sys/sysctl.h`. The program compiles just fine with AppleClang 10.0.1

--- a/include/internal/catch_debugger.cpp
+++ b/include/internal/catch_debugger.cpp
@@ -18,19 +18,23 @@
 #  include <stdbool.h>
 #  include <sys/types.h>
 #  include <unistd.h>
-#  include <sys/sysctl.h>
 #  include <cstddef>
 #  include <ostream>
 
-namespace Catch {
+#ifdef __apple_build_version__
+    // These headers will only compile with AppleClang (XCode)
+    // For other compilers (Clang, GCC, ... ) we need to exclude them
+#  include <sys/sysctl.h>
+#endif
 
+    namespace Catch {
+        #ifdef __apple_build_version__
         // The following function is taken directly from the following technical note:
-        // http://developer.apple.com/library/mac/#qa/qa2004/qa1361.html
+        // https://developer.apple.com/library/archive/qa/qa1361/_index.html
 
         // Returns true if the current process is being debugged (either
         // running under the debugger or has a debugger attached post facto).
         bool isDebuggerActive(){
-
             int                 mib[4];
             struct kinfo_proc   info;
             std::size_t         size;
@@ -60,6 +64,12 @@ namespace Catch {
 
             return ( (info.kp_proc.p_flag & P_TRACED) != 0 );
         }
+        #else
+        bool isDebuggerActive() {
+            // We need to find another way to determine this for non-appleclang compilers on macOS
+            return false;
+        }
+        #endif
     } // namespace Catch
 
 #elif defined(CATCH_PLATFORM_LINUX)

